libcxx/src/filesystem/operations.cpp

```diff
@@ -166,15 +163,15 @@ void __copy(const path& from, const path& to, copy_options options, error_code*
       return;
     }
     error_code m_ec2;
-    for (; it != directory_iterator(); it.increment(m_ec2)) {
-      if (m_ec2) {
-        return err.report(m_ec2);
-      }
+    for (; !m_ec2 && it != directory_iterator(); it.increment(m_ec2)) {
```

Comment:
I don't understand this change. IIUC, the PR description mentions that we incorrectly forgot to check `m_ec2` after the recursive call to `__copy`, but this didn't change after your patch. We're still passing `ec` to the recursive `__copy` call (not `m_ec2`). I think I am missing something.

Reply:
It's that we don't check m_ec2 after the call to 'it.increment(m_ec2)'

After the for loops increment we now check m_ec2 as the loop condition, and additionally we make sure now to check the code after the final increment whereas we wouldn't before.

Reply:
Ok, I see it now, thanks!